### ISSUE TITLE
[train] Remove Framework-specific checkpoints from docs and port to new train.Checkpoint API

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#38292]
This issue is for (1) de-emphasizing framework-specific checkpoints, and (2) porting them to the new train.Checkpoint class. The goal here is not to be 100% backwards compatible, but to provide a reasonably smooth migration for users that are depending heavily on framework-specific checkpoints.

As part of this, we should be able to completely remove the air.Checkpoint class.

Depends on: https://github.com/ray-project/ray/issues/38291

## TODO

- [x] Update Tensorflow `ReportCheckpointCallback` and possibly add a getter to retrieve the model after.
- [x] Update all tune integration callbacks to use `train.report` @krfricke 
- [x] Replace old lightning callbacks with the new one (?) @woshiyyya 
- [x] Add legacy methods `Checkpoint` and have them raise an error to suggest the new way to do things.

**Comment by woshiyyya** [comment]
Ah I think we are already using the new Checkpoint API in the new [RayTrainReportCallback](https://github.com/ray-project/ray/blob/2ffe613248f39ebea725c76a5e86dfcb2db6040d/python/ray/train/lightning/_lightning_utils.py#L192).

**Comment by justinvyu** [comment]
Oh @woshiyyya this is referring to replacing the callbacks that can be used w/ Tune only with the new one you linked. https://docs.ray.io/en/latest/ray-air/api/integrations.html#one-vanilla-pytorch-lightning-with-a-ray-callback

**Comment by anyscalesam** [comment]
Justin's PR merge #38452; this won't fix the rollout. Justin will need to cut a follow up PR to enable it for tests.
